parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -242,29 +236,37 @@ pub mod pallet {
 					config::BLOCK_ROOTS_SUBTREE_INDEX,
 					config::BLOCK_ROOTS_DEPTH,
 					update.header.state_root
-				)
-				.is_some_and(|x| x),
+				),
 				Error::<T>::InvalidBlockRootsRootMerkleProof
 			);
 
 			let sync_committee_prepared: SyncCommitteePrepared = (&update.current_sync_committee)
 				.try_into()
 				.map_err(|_| <Error<T>>::BLSPreparePublicKeysFailed)?;
 			<CurrentSyncCommittee<T>>::set(sync_committee_prepared);
+			<NextSyncCommittee<T>>::kill();
 
 			Self::store_validators_root(update.validators_root);
 			Self::store_finalized_header(header_root, update.header, update.block_roots_root)?;
 
 			Ok(())
 		}
 
-		fn process_update(update: &Update) -> DispatchResult {
+		// reference and strict follows https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/sync-protocol.md#validate_light_client_update
+		fn validate_light_client_update(update: &Update) -> DispatchResult {
+			// Verify sync committee has sufficient participants
+			let participation =
+				decompress_sync_committee_bits(update.sync_aggregate.sync_committee_bits);
+			Self::sync_committee_participation_is_supermajority(&participation)?;
+
 			// Verify update does not skip a sync committee period
 			ensure!(
-				update.signature_slot > update.attested_header.slot,
-				Error::<T>::InvalidSignatureSlot
+				update.signature_slot > update.attested_header.slot &&
+					update.attested_header.slot >= update.finalized_header.slot,
+				Error::<T>::InvalidUpdateSlot
 			);
-			let store_period = compute_period(Self::latest_finalized_header().beacon_slot);
+			let latest_state = Self::latest_finalized_header();
```

Comment:
Be aware that in my latest commits, that the `LatestFinalizedHeader` storage item has been renamed to `LatestFinalizedBlockRoot` and only contains the block root. You need to make a second lookup with the block root in `FinalizedBeaconState` to get the slot number.